extensions/script-libraries/drawing/drawing.livecodescript

```diff
@@ -0,0 +1,2960 @@
+ * the feature state of the context to their 'compiled' values. The compiled
+ * values for the features are such that equivalence can be checked by
+ * simple equality. */
+private command _svgCompileFeatures @xContext, pElement
+	repeat for each key tFeatureName in pElement["features"]
+		local tFeatureValue
+		put empty into tFeatureValue
+
+		switch tFeatureName
+		case "transform"
+			local tTransform
+			put _svgContextGetState(xContext, "transform") into tTransform
+			seqAppend tTransform, pElement["features"]["transform"]
+			_svgCompileTransform xContext, tTransform, tFeatureValue
+			break
+		case "fill"
+		case "stroke"
+			_svgCompilePaint xContext, pElement["features"][tFeatureName], tFeatureValue
+			break
+		case "fill-opacity"
+		case "fill-rule"
+		case "stroke-opacity"
+		case "stroke-width"
+		case "stroke-linejoin"
+		case "stroke-linecap"
+		case "stroke-miterlimit"
+		case "stroke-dashoffset"
+		case "stroke-dasharray"
+			put pElement["features"][tFeatureName] into tFeatureValue
+			break
+		default
+			/* Ignore any features which aren't pertinent to compilation */
+			next repeat
+		end switch
+		_svgContextSetState xContext, tFeatureName, tFeatureValue
+	end repeat
+end _svgCompileFeatures
+
+private command _svgCompileTransform @xContext, pTransform, @rCompiledTransform
+	local tWidth, tHeight, tMatrix
+	put xContext["root-width"] into tWidth
+	put xContext["root-height"] into tHeight
+	put _svgTransformIdentity() into tMatrix
+
+	repeat for each element tTransform in pTransform
+		local tNextMatrix
+		switch tTransform[1]
+		case "viewport"
+			local tVpWidth, tVpHeight, tVpX, tVpY, tVbWidth, tVbHeight, tVbX, tVbY, tPar
+			put tTransform[2] into tVpX
+			put tTransform[3] into tVpY
+			put tTransform[4] into tVpWidth
+			put tTransform[5] into tVpHeight
+			put tTransform[6] into tVbX
+			put tTransform[7] into tVbY
+			put tTransform[8] into tVbWidth
+			put tTransform[9] into tVbHeight
+			put tTransform[10] into tPar
+
+			/* Resolve percentage sizes */
+			if tVpWidth < 0 then
+				put -tVpWidth * tWidth into tVpWidth
+			end if
+			if tVpHeight < 0 then
+				put -tVpHeight * tHeight into tVpHeight
+			end if
+
+			/* If the new vp sizes are not percentages then we can build a matrix
+			 * else we must emit a separate transform */
+			if tVpWidth >= 0 and tVpHeight >= 0 then
+				local tVScaleX, tVScaleY, tVTranslateX, tVTranslateY
+				put tVpWidth / tVbWidth into tVScaleX
+				put tVpHeight / tVbHeight into tVScaleY
+				if word 2 of tPar is "meet" then
+					put min(tVScaleX, tVScaleY) into tVScaleX
+					put tVScaleX into tVScaleY
+				else if word 2 of tPar is "slice" then
+					put max(tVScaleX, tVScaleY) into tVScaleX
+					put tVScaleX into tVScaleY
+				end if
+				put tVpX - (tVbX * tVScaleX) into tVTranslateX
+				put tVpY - (tVbY * tVScaleY) into tVTranslateY
+				if tPar contains "xMid" then
+					add (tVpWidth - tVbWidth * tVScaleX) / 2 to tVTranslateX
+				end if
+				if tPar contains "xMax" then
```

Comment:
Heh, I guess I'm duty bound to suggest `else if` here too ;-)

Reply:
:)